django-stubs/db/models/query.pyi

```diff
@@ -46,7 +48,7 @@ class FlatValuesListIterable(BaseIterable):
 
 _T = TypeVar("_T", bound=models.Model, covariant=True)
 
-class QuerySet(Iterable[_T], Sized):
+class QuerySet(Generic[_T, _Row], Sized):
```

Comment:
Is it possible to preserve `QuerySet` being an `Iterable` here? Or even a `Collection`, as it implements all required methods.

Reply:
Thanks for your comments. It looks like `Collection` will work, at least for the stubs tests. Let's see if Django tests pass.

Reply:
Could you also leave `Sized` be, AFAIK `Collection` doesn't subclass it.